tutorials/optimization/using_multimesh.rst

```diff
@@ -0,0 +1,71 @@
+.. _doc_using_multimesh:
+
+Optimization using MultiMeshes
+==============================
+
+For large amount of instances (in the thousands), that need to be constantly processed
+(and certain amount of control needs to be retained),
+:ref:`using servers directly <doc_using_servers>` is the recommended optimization.
+
+When the amount of objects reach the hundreds of thousands or millions,
+none of these approaches are efficient anymore. Still, depending on the requirements, there
+is one more optimization possible.
+
+MultiMeshes
+-----------
+
+A :ref:`MultiMesh<class_MultiMesh>` is a single draw primitive that can draw up to millions
+of objects in one go. It's extremely efficient because it uses the GPU hardware to do this
+(in OpenGL ES 2.0, it's less efficient because there is no hardware support for it, though).
+
+The only drawback is that there is no *screen* or *frustum* culling possible for individual instances.
+This means, that millions of objects will be *always* or *never* drawn, depending on the visibility
+of the whole MultiMesh. It is possible to provide a custom visibility rect for them, but it will always
+be *all-or-none* visibility.
+
+If the objects are simple enough (just a couple of vertices), this is generally not much of a problem
+as most modern GPUs are optimized for this use case. A workaround is to create several MultiMeshes
+for different areas of the world.
+
+It is also possible to execute some logic inside the vertex shader (using the ``INSTANCE_ID`` or
+``INSTANCE_CUSTOM`` built-in constants). For an example of animating thousands of objects in a MultiMesh,
+see the :ref:`Animating thousands of fish <doc_animating_thousands_of_fish>` tutorial. Information
+to the shader can be provided via textures (there are floating point :ref:`Image<class_Image>` formats
+which are ideal for this).
+
+Another alternative is to use GDNative and C++, which should be extremely efficient (it's possible
+to set the entire state for all objects using linear memory via the
+:ref:`VisualServer.multimesh_set_as_bulk_array() <class_VisualServer_method_multimesh_set_as_bulk_array>`
+function). This way, the array can be created with multiple threads, then set in one call, providing
+high cache efficiency.
+
+Finally, it's not required to have all MultiMesh instances visible. The amount of visible ones can be
```

Comment:
This method was also added in master and cherrypicked for 3.1.1-stable, I'll add a note to mention it.